x-pack/plugin/vector-tile/src/main/java/org/elasticsearch/xpack/vectortile/rest/VectorTileUtils.java

```diff
@@ -64,6 +64,10 @@ public static void addPropertyToFeature(VectorTile.Tile.Feature.Builder feature,
             // guard for null values
             return;
         }
+        if (value instanceof Map<?, ?> map) {
+            // maps should have been flattened already in Maps#flatten but still contains the original maps
```

Comment:
Maps.flatten does include recursive code for this, so it is surprising it does not happen. Perhaps there is a bug in Maps.flatten.